### DISCUSSION TITLE
Running on Android fails with java.lang.ClassNotFoundException: Didn't find class "java.net.http.HttpClient"

### DISCUSSION THREAD

**Discussion by bringert** [sashirestela/simple-openai#114] [Ideas]
simple-openai depends on java.net.http.HttpClient which doesn't exist on Android. This causes any attempt to use the library on Android to fail with: `java.lang.ClassNotFoundException: Didn't find class "java.net.http.HttpClient"`

Perhaps support could be added for using okhttp instead of java.net.http.HttpClient?

**Comment by sashirestela** [comment]
@bringert Thanks for trying simple-openai. Using `okhttp` is something that I need to evaluate for future releases. I need to evaluate if I can achieve the same ergonomics and ease of use as with the Java HttpClient.

Meanwhile, I'm going to move your issue to the Discussion section.

**Comment by allentown521** [comment]
we need okhttp in android ,okhttp is better than httpclient

**Comment by sashirestela** [comment]
@bringert, @allentown521 
The latest version 3.16.0 has been updated to support OkHttp (see Readme for details). Could you check if you can use it on Android, please? Take in account that this library requires Java 11 at least.

**Comment by sashirestela** [comment]
I was able to verify that the latest version works successfully in Android
